RecoCTPPS/TotemRPLocal/plugins/CTPPSDiamondLocalTrackFitter.cc

```diff
@@ -0,0 +1,114 @@
+/****************************************************************************
+ *
+ * This is a part of CTPPS offline software.
+ * Authors:
+ *   Laurent Forthomme (laurent.forthomme@cern.ch)
+ *   Nicola Minafra (nicola.minafra@cern.ch)
+ *
+ ****************************************************************************/
+
+#include <memory>
+
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+
+#include "DataFormats/Common/interface/DetSetVector.h"
+
+#include "DataFormats/CTPPSDetId/interface/CTPPSDiamondDetId.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSDiamondRecHit.h"
+#include "DataFormats/CTPPSReco/interface/CTPPSDiamondLocalTrack.h"
+
+#include "RecoCTPPS/TotemRPLocal/interface/CTPPSDiamondTrackRecognition.h"
+
+class CTPPSDiamondLocalTrackFitter : public edm::stream::EDProducer<>
+{
+  public:
+    explicit CTPPSDiamondLocalTrackFitter( const edm::ParameterSet& );
+    ~CTPPSDiamondLocalTrackFitter();
+
+    static void fillDescriptions( edm::ConfigurationDescriptions& );
+
+  private:
+    virtual void produce( edm::Event&, const edm::EventSetup& ) override;
+
+    edm::EDGetTokenT< edm::DetSetVector<CTPPSDiamondRecHit> > recHitsToken_;
+    CTPPSDiamondTrackRecognition* trk_algo_45_;
```

Comment:
please use unique_ptr instead of raw pointers. (actually, these don't really need to be pointers at all)